Update Agent Builder built-in tools reference for 9.5 (#7262)
elastic/docs-content: explore-analyze/ai-features/agent-builder/tools/builtin-tools-reference.md

`platform.core.create_visualization` {applies_to}`stack: ga 9.4+`
:   Creates or updates a visualization configuration based on a natural language description.

> `platform.core.execute_connector_sub_action` {applies_to}`stack: preview 9.5`

Reviewer: ```suggestion
`platform.core.execute_connector_sub_action` {applies_to}`stack: preview 9.4`
```

Reviewer: Confirmed, https://github.com/elastic/kibana/pull/260784 got in for 9.4.0

`security.search_entities` {applies_to}`stack: ga 9.4+`
:   Searches the Entity store for security entities (host, user, service, or generic), with filtering by risk score, asset criticality, entity attributes, and lifecycle timestamps. Use when the entity ID (EUID) is not known, use `security.get_entity` when it is.

`security.set_asset_criticality` {applies_to}`stack: preview 9.5`

> :   Sets or removes the [asset criticality](/solutions/security/advanced-entity-analytics/asset-criticality.md) level for a security entity, which influences its risk scoring.

Reviewer 2: ```suggestion
:   Sets or removes the [asset criticality](/solutions/security/advanced-entity-analytics/asset-criticality.md) level for a security entity and recalculates the entity's risk score.
```
Per the description in https://github.com/elastic/docs-content/issues/7288.

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

### SIEM readiness tools

SIEM readiness tools assess [SIEM readiness](/solutions/security/get-started/siem-readiness.md) across four dimensions: coverage, quality, continuity, and retention.

`security.siem_readiness.get_coverage` {applies_to}`stack: preview 9.5`
:   Retrieves SIEM data coverage health across the SIEM data categories (endpoint, identity, network, cloud, and application/SaaS), including document counts, detection-rule presence, health status, and findings.

`security.siem_readiness.get_quality` {applies_to}`stack: preview 9.5`
:   Retrieves SIEM data quality health based on ECS compatibility check results, including incompatible field mappings, health status, and findings. Requires a prior Data Quality dashboard run.

`security.siem_readiness.get_continuity` {applies_to}`stack: preview 9.5`
:   Retrieves SIEM ingest pipeline continuity health, including active pipelines, failure rates, silent data streams, and volume drops.

`security.siem_readiness.get_retention` {applies_to}`stack: preview 9.5`
:   Retrieves SIEM data retention health, including data streams and indices with retention configuration, retention days, and compliance status.

> ### Watchlist tools

Reviewer 2: A comment on the groupings: watchlists and threat hunting leads are both sub-features of Entity Analytics. It might be too granular to have them under their own separate subheadings, and this also moves them away from other Entity Analytics tools (security.entity_risk_score, security.get_entity, security.search_entities, and security.set_asset_criticality), which are listed directly under Security tools.
It might be better to have an Entity Analytics tools subheading under Security tools with all the EA tools grouped together.

Author: Good call. All the Entity Analytics tools (entity_risk_score, get_entity, search_entities, set_asset_criticality, watchlists, and leads) are now grouped under a single Entity Analytics tools subheading, with watchlists and leads as labeled groups that keep their own prerequisites. Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

```{applies_to}

> stack: preview 9.5

Reviewer 2: ```suggestion
stack: ga 9.5
```

Per @tcalopes [comment](https://github.com/elastic/docs-content/pull/7333#issuecomment-4960930024).

Author: Applied as GA in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0. I removed the section-level badge (following your note below about not duplicating tags) so each tool now carries its own inline tag. I kept the `+` (`ga 9.5+`) to match the other version tags in this reference.

```

> Watchlist tools manage [Entity Analytics](/solutions/security/advanced-entity-analytics.md) watchlists. Mutating actions require user confirmation before running.

Reviewer 2: ```suggestion
Watchlist tools manage [entity watchlists](/solutions/security/advanced-entity-analytics/watchlists.md). Mutating actions require user confirmation before running.
```

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> **Prerequisites:** The `entityAnalyticsWatchlistEnabled` {{elastic-sec}} [experimental feature flag](kibana://reference/configuration-reference/security-solution-settings.md#experimental-features) must be enabled, and a [Platinum or higher license](https://www.elastic.co/subscriptions) is required. Adding and removing entities also requires Entity Store V2 (`entityAnalyticsEntityStoreV2`).

Reviewer 2: ```suggestion
**Prerequisites:** A [Platinum or higher license](https://www.elastic.co/subscriptions) is required. Adding and removing entities also requires the [entity store](/solutions/security/advanced-entity-analytics/entity-store.md) to be enabled and populated.
```

> The `entityAnalyticsWatchlistEnabled` {{elastic-sec}} [experimental feature flag](kibana://reference/configuration-reference/security-solution-settings.md#experimental-features) must be enabled 

This FF is true by default, and the watchlists feature itself is GA, so I think we can omit this.

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> `security.list_watchlists` {applies_to}`stack: preview 9.5`

Reviewer 2: We prob don't need the applies tags both at the section level and inline

Author: Done. The section-level badges are gone and each tool carries its own inline tag now. Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> :   Lists the Entity Analytics watchlists in the current space, including name, description, risk modifier, and member sources.

Reviewer 2: ```suggestion
:   Lists the entity watchlists in the current space, including name, description, risk modifier, and member sources.
```

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

`security.create_watchlist` {applies_to}`stack: preview 9.5`
:   Creates a new Entity Analytics watchlist in the current space.

`security.update_watchlist` {applies_to}`stack: preview 9.5`
:   Updates an existing watchlist, such as renaming it or changing its description or risk modifier.

`security.delete_watchlist` {applies_to}`stack: preview 9.5`
:   Permanently deletes an Entity Analytics watchlist. This action cannot be undone.

`security.add_entities_to_watchlist` {applies_to}`stack: preview 9.5`
:   Adds one or more entities to a watchlist by entity ID (EUID).

`security.remove_entities_from_watchlist` {applies_to}`stack: preview 9.5`
:   Removes one or more entities from a watchlist by entity ID (EUID).

### Lead generation tools
```{applies_to}
stack: preview 9.5
```

> Lead generation tools surface AI-generated investigation leads for security entities.

Reviewer 2: ```suggestion
Lead generation tools surface AI-generated [threat hunting leads](/solutions/security/advanced-entity-analytics/monitor-entity-risk.md#entity-threat-hunting-leads) for security entities.
```

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> **Prerequisites:** The `leadGenerationEnabled` {{elastic-sec}} [experimental feature flag](kibana://reference/configuration-reference/security-solution-settings.md#experimental-features) must be enabled, and an [Enterprise license](https://www.elastic.co/subscriptions) is required.

Reviewer 2: Same here, not sure we need to mention the FF since it's true by default?

Author: Agreed. I removed the feature flag from the prerequisites and kept the Enterprise license. Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> `security.list_leads` {applies_to}`stack: preview 9.5`

Reviewer 2: Same as https://github.com/elastic/docs-content/pull/7262#discussion_r3577375983

Author: Done, same as above. Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

> :   Lists AI-generated investigation leads for security entities, sorted by priority.

Reviewer 2: ```suggestion
:   Lists AI-generated threat hunting leads for security entities, sorted by priority.
```

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

`security.generate_leads` {applies_to}`stack: preview 9.5`

> :   Triggers AI-powered investigation lead generation. The job runs asynchronously and returns immediately.

Reviewer 2: ```suggestion
:   Generates new threat hunting leads. The job runs asynchronously and returns immediately.
```
Maybe "returns results immediately" sounds more natural?

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0. I kept "returns immediately" rather than "returns results immediately", since the job runs asynchronously and returns before the results are ready.

`security.dismiss_lead` {applies_to}`stack: preview 9.5`

> :   Dismisses an AI-generated investigation lead by ID, marking it as triaged.

Reviewer 2: ```suggestion
:   Dismisses an AI-generated threat hunting lead by ID, marking it as triaged.
```

Author: Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0

### PCI compliance tools
```{applies_to}
stack: preview 9.5
```

PCI compliance tools support PCI DSS v4.0.1 compliance assessments.

**Prerequisites:** The `pciComplianceAgentBuilder` {{elastic-sec}} [experimental feature flag](kibana://reference/configuration-reference/security-solution-settings.md#experimental-features) must be enabled.

`security.pci_scope_discovery` {applies_to}`stack: preview 9.5`
:   Discovers PCI-relevant data coverage across indices, including custom-ingested data, and classifies it by scope area.

`security.pci_compliance` {applies_to}`stack: preview 9.5`
:   Runs PCI DSS v4.0.1 compliance checks or reports, returning either per-requirement findings with ES|QL evidence or a visual scorecard.

`security.pci_field_mapper` {applies_to}`stack: preview 9.5`
:   Inspects non-ECS index fields and suggests mappings to ECS fields for PCI queries. Use when scope discovery reports low ECS coverage.

## Inline tools

> Some [built-in skills](../builtin-skills-reference.md) include inline tools that are only available while that skill is active. Because they are scoped to a skill rather than assignable on their own, they are not listed as standalone tools above. For example:

Reviewer 3: ```suggestion
Some [built-in skills](../builtin-skills-reference.md) include inline tools that are only available while that skill is active. Because they are scoped to a skill rather than assignable on their own, they are not listed among the namespaced tools in this reference. For example:
```

- The [`dashboard-management`](../builtin-skills-reference.md#agent-builder-dashboard-management-skill) skill includes an inline tool for generating and updating dashboards.

> - The `rule-management` skill includes the `platform.alerting.manage_rule` tool for composing and modifying alerting rules.

Reviewer 4: It also includes the `platform.alerting.manage_action_policy` tool (https://github.com/elastic/kibana/blob/c7a678ecd5a56a4ce131b7e4bf47136b8802a174/x-pack/platform/plugins/shared/alerting_v2/server/agent_builder/skills/rule_management_skill.ts#L507)

Author: Good catch. I added `platform.alerting.manage_action_policy` to the rule-management inline tools. Added in https://github.com/elastic/docs-content/pull/7262/commits/f61975050a123614a830dc42cc80c3d6898a88e0
